### PR TITLE
[JSON-RPC] Use int to volume in OnVolumeChanged

### LINKED ISSUE #17894: [JSON-RPC] Application.OnVolumeChanged returns float volume
### Describe the bug
Here is a clear and concise description of what the problem is:

In JSON-RPC API, the [`Application.OnVolumeChanged`](https://kodi.wiki/view/JSON-RPC_API/v10#Application.OnVolumeChanged) notification returns the `volume` in a float. The documentation indicates that it's an integer. And the [`Application.GetProperties`](https://kodi.wiki/view/JSON-RPC_API/v10#Application.GetProperties) method returns an integer.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Kodi should send the notification with an integer for the `"volume":98`.
```json
{"jsonrpc":"2.0","method":"Application.OnVolumeChanged","params":{"data":{"muted":false,"volume":98},"sender":"xbmc"}}
```

## Actual Behavior

Kodi sends the notification with an float for the `"volume":98.88888549804688`.
```json
{"jsonrpc":"2.0","method":"Application.OnVolumeChanged","params":{"data":{"muted":false,"volume":98.88888549804688},"sender":"xbmc"}}
```

## Possible Fix

As in the [`CApplicationOperations::GetPropertyValue()`](https://github.com/xbmc/xbmc/blob/654c1471a62deeec8a6c6b485f7bb2e1969924ea/xbmc/interfaces/json-rpc/ApplicationOperations.cpp#L107), cast the volume in int: **`data["volume"] = static_cast<int>(GetVolumePercent());`** (cf. [`CApplication::VolumeChanged()`](https://github.com/xbmc/xbmc/blob/654c1471a62deeec8a6c6b485f7bb2e1969924ea/xbmc/Application.cpp#L4469)).

### To Reproduce

1. In Kodi, *Allow remote control from applications on other systems* (*Settings* / *Services* / *Control*).
2. Execute (by changing the IP address) : `curl --include --no-buffer --header "Connection: Upgrade" --header "Upgrade: websocket" --header "Sec-WebSocket-Version: 13" --http0.9 --output out.log http://127.0.0.1:9090/jsonrpc/`
3. Change volume in kodi.
4. Stop `curl`.
5. Open *out.log* that contains `Application.OnVolumeChanged` notification.

### Debuglog

The debuglog can be found here:

Kodi log nothing.

### Screenshots 
Here are some links or screenshots to help explain the problem:

NA

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

NA

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [X] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: OSMC 2020.03-1
 - Kodi version: 18.6

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by regseb**
## Description

In JSON-RPC API, use `int` to `volume` property in [`Application.OnVolumeChanged`](https://kodi.wiki/view/JSON-RPC_API/v10#Application.OnVolumeChanged) notification.

## Motivation and Context

Pull request to fix https://github.com/xbmc/xbmc/issues/17894.

### Application.OnVolumeChanged
1. In Kodi, *Allow remote control from applications on other systems* (*Settings* / *Services* / *Control*).
2. Execute (by changing the IP address) : `curl --include --no-buffer --header "Connection: Upgrade" --header "Upgrade: websocket" --header "Sec-WebSocket-Version: 13"  --http0.9 --output out.log http://127.0.0.1:9090/jsonrpc/`
3. Change volume in kodi.
4. Stop `curl`.
5. Open *out.log* that contains `Application.OnVolumeChanged` notification with `volume` property in `int`.

### Application.GetProperties
1. In Kodi, *Allow remote control via HTTP* (*Settings* / *Services* / *Control*).
2. Execute (by changing the IP address) : `curl -d '{ "jsonrpc": "2.0", "method": "Application.GetProperties", "params": { "properties": ["volume"] }, "id": 1 }' -H 'Content-type: application/json' -X POST 127.0.0.1:8080/jsonrpc`
3. `volume` property is an `int`.

## Screenshots (if appropriate):
NA

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by garbear** [comment]
Should we round instead of truncate?

**Comment by regseb** [comment]
@garbear In the [`CApplicationOperations::GetPropertyValue`](https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/json-rpc/ApplicationOperations.cpp#L107) method, the volume is truncated. I copied the code to be consistent between _GetProperties_ and _OnVolumeChanged_.

If rounding is better, the volume can be rounded in both methods.

**Comment by regseb** [comment]
@garbear Do I replace `static_cast<int>(...)` by `(int)round(...)` for both methods ([`CApplication::VolumeChanged()`](https://github.com/xbmc/xbmc/blob/ba3be2006902d0b5e93f07e2d3a1fd997d996be2/xbmc/Application.cpp#L4469) and [`CApplicationOperations::GetPropertyValue()`](https://github.com/xbmc/xbmc/blob/ba3be2006902d0b5e93f07e2d3a1fd997d996be2/xbmc/interfaces/json-rpc/ApplicationOperations.cpp#L107))?

**Comment by garbear** [comment]
Best to use stdlib functions in the std namespace, so:

```c++
#include <cmath>

int volume = std::lround(...);
```

**Comment by regseb** [comment]
@garbear I used `std::lroundf()` in both methods because `GetVolumePercent()` returns a `float`.

File: xbmc/interfaces/json-rpc/ApplicationOperations.cpp [hunk 1]
```diff
@@ -104,7 +105,7 @@ JSONRPC_STATUS CApplicationOperations::Quit(const std::string &method, ITranspor
 JSONRPC_STATUS CApplicationOperations::GetPropertyValue(const std::string &property, CVariant &result)
 {
   if (property == "volume")
-    result = static_cast<int>(g_application.GetVolumePercent());
+    result = std::lroundf(g_application.GetVolumePercent());
```
Comment (phunkyfish) [inline xbmc/interfaces/json-rpc/ApplicationOperations.cpp:108]: You may still need to static cast here.

File: xbmc/Application.cpp [hunk 2]
```diff
@@ -4466,7 +4468,7 @@ float CApplication::GetVolumeRatio() const
 void CApplication::VolumeChanged()
 {
   CVariant data(CVariant::VariantTypeObject);
-  data["volume"] = GetVolumePercent();
+  data["volume"] = std::lroundf(GetVolumePercent());
```
Comment (phunkyfish) [inline xbmc/Application.cpp:4471]: You may still need to static cast to int here.

File: xbmc/interfaces/json-rpc/ApplicationOperations.cpp (hunk 1 above)
Comment (regseb) [reply xbmc/interfaces/json-rpc/ApplicationOperations.cpp]: Yes, I added `static_cast<int>(...)` to cast `long int` to `int`.

File: xbmc/Application.cpp (hunk 2 above)
Comment (regseb) [reply xbmc/Application.cpp]: @phunkyfish Yes, I added `static_cast<int>(...)` to cast `long int` to `int`.

**Comment by regseb** [comment]
@garbear @phunkyfish I corrected both feedback. That's good for you ?

**Comment by phunkyfish** [comment]
You would still need to increase the json API version. @DaveTBlake in this case should the minor or macro version be updated?

**Comment by regseb** [comment]
@phunkyfish @DaveTBlake I incremented the [patch version](https://github.com/xbmc/xbmc/pull/17916/files#diff-98ec18f1c0ae94279b15422a776a7aad8aab8c9efde1a805598739c78a2fd27c) because this pull request fixes a bug.

**Comment by DaveTBlake** [comment]
Yeap, patch bump as an internal change to implementation

**Pull request merged**